File: app/src/main/java/com/wangzhen/reader/ui/adapter/PageStyleAdapter.java
package com.wangzhen.reader.ui.adapter;

import android.graphics.drawable.Drawable;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.wangzhen.adapter.RecyclerAdapter;
import com.wangzhen.reader.ui.adapter.holder.PageStyleHolder;
import com.wangzhen.reader.widget.page.PageStyle;

import java.util.List;

/**
 * PageStyleAdapter
 * Created by wangzhen on 2023/4/13
 */
public class PageStyleAdapter extends RecyclerAdapter<Drawable> {
    private int currentChecked;

    public PageStyleAdapter(List<Drawable> list) {
        super(list);
    }

    @Override
    public RecyclerView.ViewHolder onAbsCreateViewHolder(ViewGroup parent, int viewType) {
        return new PageStyleHolder(parent);
    }

    @Override
    public void onBindViewHolder(@NonNull RecyclerView.ViewHolder holder, int position) {
        super.onBindViewHolder(holder, position);
        if (currentChecked == position && holder instanceof PageStyleHolder) {
            ((PageStyleHolder) holder).setChecked();
        }
    }

    public void setPageStyleChecked(PageStyle pageStyle) {
        currentChecked = pageStyle.ordinal();
        notifyDataSetChanged();
    }
}

File: app/src/main/java/com/wangzhen/reader/ui/adapter/PageStyleAdapter.kt
package com.wangzhen.reader.ui.adapter

import android.graphics.drawable.Drawable
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.wangzhen.adapter.RecyclerAdapter
import com.wangzhen.reader.ui.adapter.holder.PageStyleHolder
import com.wangzhen.reader.widget.page.PageStyle

/**
 * PageStyleAdapter
 * Created by wangzhen on 2023/4/13
 */
class PageStyleAdapter(list: List<Drawable>) : RecyclerAdapter<Drawable?>(list) {
    private var currentChecked = 0
    override fun onAbsCreateViewHolder(parent: ViewGroup, viewType: Int): RecyclerView.ViewHolder {
        return PageStyleHolder(parent)
    }

    override fun onBindViewHolder(holder: RecyclerView.ViewHolder, position: Int) {
        super.onBindViewHolder(holder, position)
        if (currentChecked == position && holder is PageStyleHolder) {
            holder.setChecked()
        }
    }

    fun setPageStyleChecked(pageStyle: PageStyle) {
        currentChecked = pageStyle.ordinal
        notifyDataSetChanged()
    }
}